### PR TITLE
formula_cellar_checks: check Python site packages

### PR DESCRIPTION
**Pull request by Bo98**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/master/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/master/Library/Homebrew/test/PATH_spec.rb).
- [x] Have you successfully run `brew style` with your changes locally?
- [x] Have you successfully run `brew tests` with your changes locally?

-----

This audit compares this list to the list of non-recursive dependencies.

Intended behaviours:

* Formula depends on python@3.8 and package is installed for Python 3.8 -> pass
  - If there are additional packages for other versions they are ignored. This allows formulae to supply packages for multiple Python versions.
* Python package for 3.7 installed but depends on python@3.8 -> fail
* Python package installed but no Python dependencies -> pass
  - Python packages may be an "optional" part of the formula.

When testing, dependencies are not searched recursively. It's difficult to guarantee zero false positives here, particularly if the dependency tree has mixed versions of Python.

### CODE REVIEW

File: Library/Homebrew/linkage_checker.rb [hunk 1]
```diff
@@ -233,6 +241,24 @@ def harmless_broken_link?(dylib)
     ].include?(dylib)
   end
 
+  def check_packages
```
Comment (MikeMcQuaid) [inline Library/Homebrew/linkage_checker.rb:244]: Feels like this is somewhat stretch the definition of `linkage_checker`. Could this be a post-install audit instead?

Comment (Bo98) [reply Library/Homebrew/linkage_checker.rb:244]: I think so. There was differing opinions when I asked a few weeks back so wasn't entirely sure.

File: Library/Homebrew/formula_cellar_checks.rb [hunk 2]
```diff
@@ -165,6 +165,30 @@ def check_elisp_root(share, name)
     EOS
   end
 
+  def check_python_packages(lib, deps)
+    return unless lib.directory?
+
+    lib_subdirs = lib.children.select(&:directory?).map(&:basename)
```
Comment (MikeMcQuaid) [inline Library/Homebrew/formula_cellar_checks.rb:171]:
```suggestion
    lib_subdirs = lib.children
                     .select(&:directory?)
                     .map(&:basename)
```

File: Library/Homebrew/formula_cellar_checks.rb [hunk 3]
```diff
@@ -165,6 +165,30 @@ def check_elisp_root(share, name)
...
+
+    pythons = lib_subdirs.map { |p| p.to_s.match(/^python(\d+\.\d+)$/)&.captures&.first }.compact
```
Comment (MikeMcQuaid) [inline Library/Homebrew/formula_cellar_checks.rb:173]:
```suggestion
    pythons = lib_subdirs.map do |p| 
      match = p.to_s.match(/^python(\d+\.\d+)$/)
      next if match.blank?
      next if match.captures.blank?
      
      match.captures.first
    end.compact
```

File: Library/Homebrew/formula_cellar_checks.rb [hunk 4]
```diff
@@ -165,6 +165,30 @@ def check_elisp_root(share, name)
...
+    python_deps = deps.map(&:name)
+                      .grep(/^python(@.*)?$/)
+                      .map { |d| Formulary.factory(d).version.to_s[/^\d+\.\d+/] }
```
Comment (MikeMcQuaid) [inline Library/Homebrew/formula_cellar_checks.rb:176]:
```suggestion
                      .map { |d| Formula[d].version.to_s[/^\d+\.\d+/] }
```

File: Library/Homebrew/formula_cellar_checks.rb [hunk 5]
```diff
@@ -165,6 +165,30 @@ def check_elisp_root(share, name)
...
+                      .compact
+
+    return unless !python_deps.empty? && pythons.none? { |v| python_deps.include? v }
```
Comment (MikeMcQuaid) [inline Library/Homebrew/formula_cellar_checks.rb:179]:
```suggestion
    return if python_deps.blank?
    return if pythons.none? { |v| python_deps.include? v }
```

**Comment by MikeMcQuaid** [comment]
Thanks again @Bo98!

**Pull request merged**